src/Makefile

```diff
@@ -530,5 +540,24 @@ install: all
 	@ln -sf $(SERVER_NAME) $(INSTALL_BIN)/$(ENGINE_CHECK_AOF_NAME)
 	@ln -sf $(SERVER_NAME) $(INSTALL_BIN)/$(ENGINE_SENTINEL_NAME)
 
+# Create some symlinks like `redis-server` link to `valkey-server` for compatibility
+ifeq ($(CONFIGURE_REDIS_SYMLINKS), true)
+	$(call CREATE_SYMLINK,$(SERVER_NAME),$(INSTALL_BIN),$(subst $(ENGINE_NAME),redis,$(SERVER_NAME)))
```

Comment:
It's difficult to see that this is inside the body of the `make install` target.

We need to rewrite it in some way so the indentation looks intuitive.

Reply:
Can we define CREATE_SYMLINK conditionally to either create a symlink or be a no-op? In that case, maybe we should rename it to MAYBE_INSTALL_REDIS_SYMLINK instead.

Then we can just call it inside the install target unconditionally:

```make
install: all
	...
	@$(call CREATE_SYMLINK,$(ENGINE_CLI_NAME),$(INSTALL_BIN),$(subst $(ENGINE_NAME),redis,$(ENGINE_CLI_NAME)))

Reply:
Your suggestion makes the code look clearer.
I created the `define MAYBE_INSTALL_REDIS_SYMLINK` to handler whether to create redis symlinks. 

`make install`  worked well and `make V=1 install` worked like this:
```shell
cd src && /Library/Developer/CommandLineTools/usr/bin/make install

Hint: It's a good idea to run 'make test' ;)

install valkey-server /usr/local/bin
install valkey-benchmark /usr/local/bin
install valkey-cli /usr/local/bin
if [ yes = "yes" ]; then         ln -sf valkey-server /usr/local/bin/redis-server; fi
if [ yes = "yes" ]; then         ln -sf valkey-cli /usr/local/bin/redis-cli; fi
if [ yes = "yes" ]; then         ln -sf valkey-benchmark /usr/local/bin/redis-benchmark; fi
if [ yes = "yes" ]; then         ln -sf valkey-check-rdb /usr/local/bin/redis-check-rdb; fi
if [ yes = "yes" ]; then         ln -sf valkey-check-aof /usr/local/bin/redis-check-aof; fi
if [ yes = "yes" ]; then         ln -sf valkey-sentinel /usr/local/bin/redis-sentinel; fi
```

Is there any way to hide the if code and only show ` ln -sf valkey-server /usr/local/bin/redis-server;`?
Looking forward to your reply.

Reply:
Yes, here you do the `if` in the shell. We can do it in make. I'll show you. One second.

Reply:
The suggestions I posted below are untested, but I edited them afterwards because I realized I forgot some dollar signs...

Here is a makefile example that I actually tested:

```make
USE_SYMLINKS?=yes

ifeq ($(USE_SYMLINKS),yes)
    MAYBE_INSTALL_SYMLINK=$(INSTALL_SYMLINK)
else
    MAYBE_INSTALL_SYMLINK=
endif

INSTALL_SYMLINK=@echo Here we would do this: ln -s $1 $2

install:
	$(call MAYBE_INSTALL_SYMLINK,link,target)

.PHONY: install
```